packages/admin/src/pages/super-admin-dashboard/user/[id]/change-roles.test.tsx

```diff
@@ -1,8 +1,25 @@
 import '@testing-library/jest-dom';
+import { parseBody } from 'next/dist/server/api-utils/node';
 import { render, screen } from '@testing-library/react';
-import EditRoleWithId from './change-roles.page';
+import EditRoleWithId, { getServerSideProps } from './change-roles.page';
+import UserDetails from '../../../../types/UserDetails';
+import { getContext } from 'gap-web-ui';
+import { getUserById } from '../../../../services/SuperAdminService';
 import { User } from '../../types';
 
+jest.mock('../../../../services/SuperAdminService', () => ({
+  getUserById: jest.fn(),
+  getAllRoles: async () => getMockRoles(),
+  updateUserRoles: jest.fn(),
+}));
+
+jest.mock('next/dist/server/api-utils/node', () => {
+  return { parseBody: jest.fn() };
+});
```

Comment:
more nitpicking but you  can use implicit return here like in the function above:

```suggestion
jest.mock('next/dist/server/api-utils/node', () => ({
  parseBody: jest.fn(),
);
```